packages/peregrine/lib/util/createProductVariants.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Rebuild the array of variants with out of stock items data added.
+ * Since when admin selects in the Admin dashboard to not to display out of stock products
+ * the variants data that are needed to find disabled swatches only show the the in stock ones, missing the out of stock ones
+ * We rebuild the variants here to display all the variants and mark the stock status accordingly
+ * This returns an array of objects
+ */
+
+export const createProductVariants = product => {
+    const OUT_OF_STOCK_CODE = 'OUT_OF_STOCK';
+    const IN_STOCK_CODE = 'IN_STOCK';
+
+    if (product && product.configurable_options) {
+        const { variants } = product;
+        // Compute the permutation of all possible arrays of given arrays
+        // For example, if array = [[1,2],[10,20],[100,200,300]]
+        // the result is [[1, 10, 100], [1, 10, 200], [1, 10, 300], [1, 20, 100], [1, 20, 200],
+        // [1, 20, 300], [2, 10, 100], [2, 10, 200], [2, 10, 300], [2, 20, 100], [2, 20, 200], [2, 20, 300]]
+        const cartesian = (...array) =>
+            array.reduce((array, current) =>
+                array.flatMap(cur => current.map(n => [cur, n].flat()))
+            );
+
+        let configurableOptionsValueIndexes = product.configurable_options.map(
+            option => option.values.map(value => value.value_index)
+        );
+        // Get all possible variants for current options
+        let allPossibleItems = cartesian(...configurableOptionsValueIndexes);
+
+        let variantsValueIndexes = variants.map(variant =>
+            variant.attributes.map(attribute => attribute.value_index)
+        );
+
+        let newVariantsArray = [];
+        let len = allPossibleItems.length;
+        let foundMatch;
+        let currentValueIndex = [];
+        //let attmp = [];
```

Comment:
don't forget to remove commented out code

Reply:
got it, It's updated. I will double check it going forward, thanks!